### PR TITLE
Cleanup HAL Download and Update Download Link

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
Everytime we build firmware on CI, we download a 932mb file that contains a bunch of example probjects, videos, tools and other random stuff that we don't use. I've removed all of that, which brings the size down to ~200mb, and re-hosted it on our new `Software-External-Dependencies` repo for a more reliable/faster download.

### Testing Done
Tested locally, fw builds + runs

### Resolved Issues
#1427 

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware_new/rules.bzl [hunk 1]
```diff
@@ -1800,19 +1536,6 @@ def stm32h7_hal_library_files_genrule(name):
             "Middlewares/Third_Party/FatFs/src/drivers/usbh_diskio_template.h",
             "Middlewares/Third_Party/FatFs/src/drivers/sd_diskio_dma_template.h",
             "Middlewares/Third_Party/FatFs/src/ff.h",
-            "Middlewares/Third_Party/LibJPEG/include/jmorecfg_template.h",
```
Comment (jonathanlew) [inline src/firmware_new/rules.bzl]: just curious why these files aren't needed

Comment (akhilveeraghanta) [reply src/firmware_new/rules.bzl]: we don't intend on manipulating images on the stm32/run the gui tools w/ the LCD so I just removed the extra stuff that we _for sure_ will not need

also I _think_ it reduces our code size, but I'm not sure if the build system is smart about unused headers

Comment (akhilveeraghanta) [reply src/firmware_new/rules.bzl]: ok nvm the unused headers are optimized away, this change is just to reduce clutter then

**Review by jonathanlew — DISMISSED** [review]

**Review by jonathanlew — COMMENTED** [review]
Just thought of something. Can we also use the old link as a backup?

**Comment by akhilveeraghanta** [comment]
closing this for now, will get around to doing this properly later

**Pull request closed without merging**